src/frontend/bytecomp/vminstdef.yaml

```diff
@@ -2622,3 +2622,33 @@ params:
 
 code: |
   raise (report_dynamic_error msg)
+
+---
+inst: ReadFile
+is-pdf-mode-primitive : yes
+is-text-mode-primitive : yes
+name: "read-file"
+type: |
+  ~% (tS @-> tL(tS))
+
+params:
+  - relpath : string
+
+code: |
+  let path =
+    Filename.concat (OptionState.job_directory ()) relpath
```

Comment:
Is there a reason to support only relative paths and not to support absolute paths? IMHO it will be more convenient to allow absolute paths.

Reply:
It is difficult to determine which is better.
I think it's best to implement both.